src/qtcore/qml/elements/Component.js

```diff
@@ -7,6 +7,20 @@ function QMLContext() {
     }
 }
 
+QMLComponent.getAttachedObject = function() { // static
+    if (!this.Component) {
```

Comment:
We need `this.$Component` here. That's because `this.Component` is a getter-setter-property and accessing that would result in this very function to be called again, causing an infinite loop ending in a stack-overflow.